csrc/mmdeploy/preprocess/transform/transform.h

```diff
@@ -6,79 +6,35 @@
 #include "mmdeploy/core/device.h"
 #include "mmdeploy/core/module.h"
 #include "mmdeploy/core/registry.h"
+#include "mmdeploy/operation/operation.h"
 
 namespace mmdeploy {
 
 using namespace framework;
 
-class MMDEPLOY_API TransformImpl : public Module {
- public:
-  TransformImpl() = default;
-  explicit TransformImpl(const Value& args);
-  ~TransformImpl() override = default;
-
- protected:
-  std::vector<std::string> GetImageFields(const Value& input);
+namespace transform {
 
- protected:
-  Device device_;
-  Stream stream_;
-};
-
-class MMDEPLOY_API Transform : public Module {
+class MMDEPLOY_API Transform {
  public:
-  ~Transform() override = default;
+  virtual ~Transform() = default;
+  virtual Result<void> Apply(Value& data) = 0;
+};
 
-  Transform() = default;
-  explicit Transform(const Value& args);
-  Transform(const Transform&) = delete;
-  Transform& operator=(const Transform&) = delete;
+MMDEPLOY_API std::vector<std::string> GetImageFields(const Value& input);
 
-  const std::string& RuntimePlatform() const { return runtime_platform_; }
+MMDEPLOY_DECLARE_REGISTRY(Transform, std::unique_ptr<Transform>(const Value& config));
 
- protected:
-  template <typename T>
-  [[deprecated]] std::unique_ptr<T> Instantiate(const char* transform_type, const Value& args,
-                                                int version = 0) {
-    std::unique_ptr<T> impl;
-    auto impl_creator = gRegistry<T>().Get(specified_platform_, version);
-    if (nullptr == impl_creator) {
-      MMDEPLOY_WARN("Cannot find {} implementation on platform {}", transform_type,
-                    specified_platform_);
-      for (auto& name : candidate_platforms_) {
-        impl_creator = gRegistry<T>().Get(name);
-        if (impl_creator) {
-          MMDEPLOY_INFO("Fallback {} implementation to platform {}", transform_type, name);
-          break;
-        }
-      }
-    }
-    if (nullptr == impl_creator) {
-      MMDEPLOY_ERROR("cannot find {} implementation on any registered platform ", transform_type);
-      return nullptr;
-    } else {
-      return impl_creator->Create(args);
-    }
-  }
+#define MMDEPLOY_REGISTER_TRANSFORM2(type, desc)                                                   \
+  MMDEPLOY_REGISTER_FACTORY_FUNC(::mmdeploy::transform::Transform, desc, [](const Value& config) { \
+    return std::make_unique<type>(config);                                                         \
+  });
 
- protected:
-  std::string specified_platform_;
-  std::string runtime_platform_;
-  std::vector<std::string> candidate_platforms_;
-};
+#define MMDEPLOY_REGISTER_TRANSFORM(type) MMDEPLOY_REGISTER_TRANSFORM2(type, (type, 0))
```

Comment:
Why not put `MMDEPLOY_REISTER_TRANSFORM` and `MMDEPLOY_REGISTER_TRANSFORM2` together?